react/features/rtcstats/RTCStats.ts

```diff
@@ -77,17 +112,21 @@ class RTCStats {
             useLegacy
         };
 
-        const rtcstatsOptions = {
-            connectionFilter,
-            pollInterval,
-            useLegacy,
-            sendSdp,
-            eventCallback: this.handleRtcstatsEvent.bind(this)
-        };
-
         this.trace = traceInit(traceOptions);
-        rtcstatsInit(this.trace, rtcstatsOptions);
-        this.initialized = true;
+        if (!this.isPeerConnectionWrapped) {
+            // FIXME we cannot unwrap the peer connection, so this has to be done once.
+            const rtcstatsOptions = {
+                connectionFilter,
+                pollInterval,
+                useLegacy,
+                sendSdp,
+                eventCallback: this.handleRtcstatsEvent.bind(this)
+            };
+
+            rtcstatsInit(this, rtcstatsOptions);
```

Comment:
I don't understand how it works that you no longer pass a reference to `trace()` to the class here instead.

Reply:
If you look [here](https://github.com/jitsi/rtcstats/blob/046450a75d60636aeedcbb77e8d79a3d7949fe86/rtcstats.js#L138), the `rtcstatsInit` function can be called with _any_ object that has a `statsEntry` method, it doesn't have to be the `trace` object. With my latest changes we pass a locally constructed object that has a `statsEntry` method and is correctly setup to reference `this.trace`.